src/resolvers/users.ts

```diff
@@ -111,6 +117,47 @@ const Query = {
   },
 };
 
+const UserMutations = {
+  /**
+   * Gives the user the experience of the completed quest
+   *
+   * @param parent - this is the return value of the resolver for this field's parent
+   * @param args - contains all GraphQL arguments provided for this field
+   * @param context - this object is shared across all resolvers that execute for a particular operation
+   */
+  async completeQuest(
+    parent: undefined,
+    { questId }: { questId: string },
+    { collection, user }: ResolverContextBase
+  ): Promise<UpdateMutationPayload<UserDBScheme>> {
+    console.log('questId');
+
+    const completedQuest = await collection('quests').findOne({ _id: new ObjectId(questId) });
+
+    if (!completedQuest) {
+      throw new UserInputError('There is no quest with such id: ' + questId);
+    }
+
+    const updatedUser = await collection('users').findOneAndUpdate(
+      { _id: new ObjectId(user.id) },
+      {
+        $inc: {
+          exp: completedQuest?.earnedExp,
+        },
+      },
+      { returnOriginal: false });
+
+    if (!updatedUser.value) {
+      throw new InvalidAccessToken();
+    }
+
+    return {
+      recordId: new ObjectId(user.id),
```

Comment:
Тут разве не просто

```suggestion
      recordId: user._id,
```
?